server/controllers/users.js

```diff
@@ -1,140 +1,159 @@
+        if (userFound) {
+          let field;
+          if (userFound.username.toUpperCase === username.toUpperCase) {
+            field = 'Username';
+          } else if (userFound.email === email) {
+            field = 'Email';
           }
-        ]
-      }
-    })
-    .then((userFound) => {
-      if (userFound) {
-        let field;
-        if (userFound.username.toUpperCase === username.toUpperCase) {
-          field = 'Username';
-        } else if (userFound.email === email) {
-          field = 'Email';
+
+          return res.status(403).json({
+            success: false,
+            message: `${field} already taken!`
+          });
         }
 
-        return res.status(403).json({
-          success: false,
-          message: `${field} already taken!`
-        });
-      }
-
-      user
-        .create({
-          name,
-          username,
-          email,
-          password: encryption.generateHash(password),
-        })
-        .then((result) => {
-          const token = auth.sign(result.id);
-
-          const createdUser = {
-            userId: result.id,
-            username: result.username,
-            email: result.email,
-            token
-          };
+        user
+          .create({
+            name,
+            username,
+            email,
+            password: newEncryption.generateHash(password),
+          })
+          .then((result) => {
+            const token = newAuth.sign(result.id);
+
+            const createdUser = {
+              userId: result.id,
+              username: result.username,
+              email: result.email,
+              token
+            };
+
+            return res.status(201).json({
+              success: true,
+              data: createdUser });
+          });
+      })
+      .catch(e => res.status(503).json({
+        success: false,
+        message: `Error Creating user ${e.message}}` }));
+
+    return this;
+  }
+
+  /**
+   * Sign In a user (Search for user)
+   *
+   * @param {object} req - HTTP Request
+   * @param {object} res - HTTP Response
+   * @returns {object} Class instance
+   * @memberof User
+   */
+  signIn(req, res) {
+    const usernameOrEmail =
+      trimWhiteSpaces(req.body.email || req.body.username);
+
+    user
+      .findOne({
+        attributes: ['id', 'name', 'username', 'email', 'password'],
+        where: {
+          $or: [
+            { username: {
+              $iLike: usernameOrEmail }
+            },
+            { email: {
+              $iLike: usernameOrEmail }
+            }
+          ]
+        }
+      })
+      .then((userFound) => {
+        if (!userFound) {
+          return res.status(404).json({
+            success: false,
+            message: 'Username or email does not exist!'
+          });
+        }
+
+        if (newEncryption.verifyHash(req.body.password, userFound.password)) {
+          const token = newAuth.sign(userFound.id);
 
           return res.status(201).json({
```

Comment:
Lookup differences between Http response code `200` and `201`; decide using the right Http response code

Reply:
okay